aws/resource_aws_lb_target_group.go

```diff
@@ -152,6 +152,16 @@ func resourceAwsLbTargetGroup() *schema.Resource {
 								"lb_cookie", // Only for ALBs
 								"source_ip", // Only for NLBs
 							}, false),
+							DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+								switch d.Get("protocol").(string) {
+								case elbv2.ProtocolEnumTcp, elbv2.ProtocolEnumUdp, elbv2.ProtocolEnumTcpUdp, elbv2.ProtocolEnumTls:
+									if new == "lb_cookie" && !d.Get("stickiness.0.enabled").(bool) {
+										log.Printf("[WARN] invalid configuration, this will fail in a future version: stickiness enabled %v, protocol %s, type %s", d.Get("stickiness.0.enabled").(bool), d.Get("protocol").(string), new)
+										return true
+									}
+								}
+								return false
+							},
```

Comment:
This would be a great place to use `ValidateDiagFunc` instead of combining `ValidateFunc` and `DiffSuppressFunc`, since it can also return warnings. It also doesn't require the practitioner to enable logging in Terraform

Reply:
The one deal killer is the need to access the `protocol`. I don't see a way to do that from the `ValidateDiagFunc`.

Reply:
Also, the `DiffSuppressFunc` should go away soon.